helsingborg-stad/app-mitt-helsingborg
# Refactor/fastlane

## Explain the changes you’ve made

Made a few tweaks to CI builds/Fastlane:

* Move generic functionality out from `/ios/fastfile` and into top-level `/fastlane` directory (will be used for Android in the future)
* Slight change to changelog text
* Change to use newer "Apple" certs (as opposed to old "iOS" certs)
  * `ios-build.yml` has been updated to use `BUILD_CERTIFICATE_BASE64_v20220725` and `P12_PASSWORD_v20220725` instead to avoid breaking changes
* Added a few extra gems useful for local development (does not affect CI)
  * Some formatting/auto-refactoring has been applied as a result

## Explain why these changes are made

Mostly prep for Android CI builds.

## How to test

Check over the changes and see if anything sticks out. Builds are already passing so no functionality has been broken as far as I see.

## Tested environments

- [x] GitHub CI
- [x] Local environment

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?
